### ISSUE TITLE
Accessibility violations

### ISSUE THREAD

**Issue by lauzadis** [Kotlin/dokka#3414]
**Is your feature request related to a problem? Please describe**
We were recently made aware that the documentation generated by Dokka violates various accessibility standards.

1. Ensure all interactive functionality is operable with the keyboard: [W3C's SC 2.1.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/keyboard-operation-keyboard-operable.html)
    - Arrow icons to expand/collapse side menu sections
    - "X" button to clear the search field
2. Ensure custom controls provide proper textual name, role, and state information: [W3C's SC 4.1.2](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html)
    - X button to clear the search field
    - Arrow icons to expand/collapse side menu sections lack name, role, state information
3. Provide a mechanism for skipping past repetitive content [W3C's SC 2.4.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-skip.html)
    - Side menu items can not be skipped. Our package has over 300 modules and navigating with the Tab key means touching each of these items before reaching the main content.
4. Ensure keyboard focus is indicated visually [W3C's SC 2.4.7](https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html)
    - "Types" and "Functions" filter buttons don't have an outline.
    - **I will submit a PR for this**
6. Ensure pages reflow without requiring two-dimensional scrolling without loss of content or functionality [W3C's SC 1.4.10](https://www.w3.org/WAI/WCAG21/Understanding/reflow.html) 
    - Links and code blocks are not properly wrapped, leading to two-dimensional scrollbar appearing on small screens. Reflow style can be added to make this wrap properly.
    - **I will submit a PR for this**

**Describe the solution you'd like**
It would be great if these violations could be addressed, it's especially useful for screen reader users.

**Describe alternatives you've considered**
None

**Additional context**
AWS SDK for Kotlin has fixed a variety of these accessibility issues by customizing Dokka's HTML / CSS files, but it would be good to get this upstreamed. I will be submitting a number of PRs which address these violations, you can take them if you like, or come up with your own solutions.

Here is a link to our docs where you can see the various violations listed: https://sdk.amazonaws.com/kotlin/api/1.0.17/index.html

**Are you willing to provide a PR?**
For some of them, yes. Some are more difficult to solve (such as skipping past repetitive content) and would require design discussion.

**Comment by IgnatBeresnev** [comment]
Thank you for the report and the contributions, Matas! Much appreciated!

The team is now actually working on addressing a bunch of UI/UX-related issues at large for the stable release of Dokka, so some if these might be covered in the updated design, but if they are not - we should definitely consider it and keep it in mind. cc @paveluxd, @d-ambatenne

**Comment by ianbotsf** [comment]
@IgnatBeresnev is there an update on when the new UI designs will be released and the WCAG violations noted above will be fixed?

**Comment by whyoleg** [comment]
Fixed by #4078, #4063, #4127, #4104
